### ISSUE TITLE
Stuck building for 10mins +

### ISSUE THREAD

**Issue by mmahalwy** [electron-userland/electron-packager#213]
I am using this alongside with ember-electron package.

I found my build is taking over 12mins where i give up and cancel it. Stuck here:

```
$ electron-packager . FooBar --platform=darwin --arch=x64 --version=0.35.0
Packaging app for platform darwin x64 using electron v0.35.0
```
Nothing happens after that.

**Comment by mmahalwy** [comment]
How can I debug this?

**Comment by malept** [comment]
Per the [contributing document](https://github.com/maxogden/electron-packager/blob/master/CONTRIBUTING.md):

> What platform are you running electron-packager on?

As to your question:

> How can I debug this? 

You can set the `DEBUG` environment variable to the value `electron-download,extract-zip`, to see if the problem lies with either of those dependencies.

**Comment by Davidaredding** [comment]
Having the same issue, identical command running on OSX 10.11.1

**Comment by malept** [comment]
@Davidaredding does using the `DEBUG` environment variable as described above produce any output in the terminal?

**Comment by Davidaredding** [comment]
Unfortunately it doesn't :(

Davids-MacBook-Pro:SDE David$ set DEBUG=electron-download,extract-zip
Davids-MacBook-Pro:SDE David$ electron-packager . FooBar --platform=darwin --arch=x64 --version=0.35.0
Packaging app for platform darwin x64 using electron v0.35.0

And there it sits.

**Comment by malept** [comment]
@Davidaredding Is that the default OSX shell? Shouldn't you be using `export` and not `set`?

**Comment by Davidaredding** [comment]
Doh'  You are correct! Sorry about that.  I do see quite a bit being written to the console now.  This seems to be the where it stops

  extract-zip mkdirp +0ms { dir: '/var/folders/k7/tqx9n0_j53v2z5y2dz1nk0vm0000gn/T/electron-packager/darwin-x64-template/Electron.app/Contents/Resources/zh_TW.lproj/' }
  extract-zip finished processing +0ms Electron.app/Contents/Resources/zh_TW.lproj/ { err: undefined }
  extract-zip zip extraction complete +0ms

**Comment by malept** [comment]
@Davidaredding I think that means that it's hanging in the Mac-specific code. What happens if you try to build an app for another OS, like linux-x64?

**Comment by Davidaredding** [comment]
seems to stop here

  extract-zip mkdirp +0ms { dir: '/var/folders/k7/tqx9n0_j53v2z5y2dz1nk0vm0000gn/T/electron-packager/linux-x64-template/locales' }
  extract-zip opening read stream +0ms /var/folders/k7/tqx9n0_j53v2z5y2dz1nk0vm0000gn/T/electron-packager/linux-x64-template/locales/ta.pak
  extract-zip finished processing +1ms locales/ta.pak { err: undefined }
  extract-zip zip extraction complete +0ms

**Comment by maxogden** [comment]
Random guess but I bet it's related to folder copying (https://github.com/maxogden/electron-packager/issues/151)

**Comment by Davidaredding** [comment]
@mmahalwy;  Were you able to get this working?

**Comment by Davidaredding** [comment]
@maxogden If this is related, what would be the potential work around?

**Comment by Davidaredding** [comment]
Also, what additional information is needed for this issue?  Happy to provide anything that will bring a speedy resolution.

**Comment by malept** [comment]
Try building on Linux. From what I can read on the other related bugs, I haven't seen any confirmation of this behavior on non-OSX machines.

**Comment by Davidaredding** [comment]
I have,  About 5 comments up.

**Comment by malept** [comment]
That was building *on* Linux, *for* any target platform?

(Side note: The electron-packager docs, including CONTRIBUTING.md, should be better at differentiating between build platforms and target platforms.)

**Comment by Davidaredding** [comment]
ah sorry, no not *on* linux, i misread your comment.  However I don't have the resources to setup a linux machine to build my project.  Is this issue strictly related to (and reproducible on) OSX?

**Comment by malept** [comment]
That's what I'm trying to ascertain. I've never had this problem, and I run exclusively Linux.

Building on a (Vagrant) VM or (Docker) container would be sufficient testing, if you're comfortable with either technology and don't want to dedicate an entire physical box.

**Comment by maxogden** [comment]
The best way to bring this to a speedy resolution would be to make a reproducible test case, e.g. code someone else can run that makes electron-packager hang at the same step during the build. It's very likely that it is something about the electron application being packaged that is causing the issue, as all of our test cases and existing users haven't hit this problem with other applications.

Basically we can't do anything to fix it without being able to reproduce the issue.

**Comment by Davidaredding** [comment]
@malept ;  Unfortunately i'm not very well versed in either technology.  

@maxogden I will attempt to reproduce it with a bare-bones electron application.  In the meantime, are there any parts of an electron application that electron-packager is dependent on that would cause this issue (or any issue) that i should look for? or this this likeliness just a shot in the dark?

**Comment by maxogden** [comment]
@Davidaredding I'm not really sure what might cause the issue. Maybe a huge number of files? All electron-packager really does is copy files from a source location into a destination and rename some stuff.

**Comment by bearjaws** [comment]
If you have the package being built into your project directory (i.e. project/build/MyApp-darwin-x64/), make sure you have package ignored the directory that it's being built into (i.e. build), otherwise it takes even longer to package (double the work)

**Comment by Ryanmtate** [comment]
I recently encountered the same problem when working on a shared repository and building on an OS X machine.

The solution was removing the .DS_Store file from the application directory root before building.

see: https://en.wikipedia.org/wiki/.DS_Store#Problems

**Comment by Ryanmtate** [comment]
Issue ultimately ended up being too many files in a bower component..

I might suggest avoid using material-design-icons bower package in electron...

**Comment by xaph** [comment]
I'm trying to build an electron app on docker with root user but it seems my build stucks also.

I'm using ubuntu 14.04 on docker.

Is there any problem on running electron-packager by root?

**Comment by mmahalwy** [comment]
@Davidaredding have you had any luck with this? Doing electron was experimental for me and I hoped this would get me up and running really quickly, but decided to put electron on the side for now. I will be revisiting in a week. Hopefully this is working?

**Comment by mmahalwy** [comment]
@xaph i have the same problem now with Docker. Did you get a fix?

**Comment by ide** [comment]
I hit this issue too with electron-packager 5.2.1 on OS X El Capitan. Adding console.log statements showed that the problem was one of the `ncp` calls here: https://github.com/electron-userland/electron-packager/blob/85410316daa219d7e08b0d3d6fb4af7ff1259efa/common.js#L83.

Restarting my computer didn't help, and I was able to build successfully with this version of electron-packager before. The two things that changed before things broke were that I had run `git clean -dfx` and npm installed again, and I updated some npm dependencies (moved from babel-polyfill to babel-runtime... should be innocuous).

I upgraded to electron-packager 6.0.0 and the issue went away for now... agree that moving away from ncp is probably a good idea. Even if it doesn't fix the issue at hand, ncp is almost totally unmaintained.

**Comment by malept** [comment]
> agree that moving away from ncp is probably a good idea. Even if it doesn't fix the issue at hand, ncp is almost totally unmaintained. 

At this point, we're looking for a community member to provide a pull request to resolve #151.

**Comment by stoufa88** [comment]
I still have an issue on osx El Capitan when packaging for Darwin. 

In fact, packager creates a directory called electron-packager, in which it runs infinite file creation loops. For example, app will have an app which also have an app... 

This only happens when packaging on osx, because it works perfectly when packaging for osx from Linux.

**Comment by averissimo** [comment]
@mmahalwy @xaph did you guys get a fix with the docker image?
